File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.kt
package org.example

fun main() {
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val people = mutableListOf<Pair<Int, String>>()
    repeat(n) {
        val (age, name) = br.readLine().split(" ")
        people.add(age.toInt() to name)
    }
    people.sortWith(compareBy({ it.first }))
    for ((age, name) in people) {
        sb.append(age).append(" ").append(name).append("\n")
    }
    println(sb)
    br.close()
}

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
import sys

input = sys.stdin.readline

n = int(input())
list = []
for i in range(n):
    age, name = map(str, input().split())
    list.append([i, int(age), name])
list.sort(key = lambda x : (x[1], x[0]))
for person in list:
    _, age, name = person
    print(age, name)